pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/DataTypeTransformer.java

```diff
@@ -73,10 +80,15 @@ public GenericRow transform(GenericRow record) {
     for (Map.Entry<String, PinotDataType> entry : _dataTypes.entrySet()) {
       String column = entry.getKey();
       Object value = record.getValue(column);
-
-      // Convert List value to Object[]
-      if (value instanceof List) {
-        value = ((List) value).toArray();
+      if (value == null) {
```

Comment:
There is no need to have an extra check here since it's been covered in Line 89.

Reply:
Perform this check before standardizing the value to prevent the extra `record.putValue()`